lesson17/index.js

```diff
@@ -0,0 +1,171 @@
+  prevBtn.classList.add("sliderBtn","prev", "is-disabled");
+  prevBtn.id = "js-slider-prevBtn";
+  const nextBtn = document.createElement("div");
+  nextBtn.classList.add("sliderBtn","next");
+  nextBtn.id = "js-slider-nextBtn";
+  main.querySelector(".slider").appendChild(prevBtn);
+  main.querySelector(".slider").appendChild(nextBtn);
+  setBtnEvent();
+}
+
+const controlBtnBehavior = () => {
+  const allSlides = Array.from(document.querySelectorAll(".sliderImg"));
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  const prevBtn = document.getElementById("js-slider-prevBtn");
+  const nextBtn = document.getElementById("js-slider-nextBtn");
+  const currentSlideIndex = allSlides.findIndex((slide)=>{
+    return slide === currentSlide;
+  });
+
+  if (currentSlideIndex === 0) {
+    prevBtn.classList.add("is-disabled");
+    nextBtn.classList.remove("is-disabled");
+  } else if(currentSlideIndex === 4) {
+    prevBtn.classList.remove("is-disabled");
+    nextBtn.classList.add("is-disabled");
+  } else {
+    prevBtn.classList.remove("is-disabled");
+    nextBtn.classList.remove("is-disabled");
+  }
+}
+
+const setFraction = (slides) => {
+  const slider = document.getElementById("js-slider");
+  const fraction = document.createElement("div");
+  fraction.classList.add("fraction");
+  const fractionWrapper = document.createElement("div");
+
+  const numerator = document.createElement("span");
+  numerator.id = "js-numerator";
+  numerator.textContent = "1";
+  const separator = document.createElement("span");
+  separator.textContent = "/";
+  const denominator = document.createElement("span");
+  denominator.textContent = slides.length;
+
+  fractionWrapper.appendChild(numerator);
+  fractionWrapper.appendChild(separator);
+  fractionWrapper.appendChild(denominator);
+
+  fraction.appendChild(fractionWrapper);
+  slider.appendChild(fraction);
+}
+
+const updateFractionNum = ()=> {
+  const allSlides = Array.from(document.querySelectorAll(".sliderImg"));
+  const currentSlide = document.querySelector('[data-hidden="false"]');
+  const currentSlideIndex = allSlides.findIndex((slide)=>{
+    return slide === currentSlide;
+  })
+  const numerator = document.getElementById("js-numerator");
+  const currentSlideNum = currentSlideIndex + 1;
+  numerator.textContent = currentSlideNum;
+}
+
+const setLoadingImage = () => {
+  const fragmentLoadingImage = document.createDocumentFragment();
+  const div = document.createElement("div");
+  div.id = "js-loading";
+  const img = document.createElement("img");
+  img.src = "./loading-circle.gif";
+  fragmentLoadingImage.appendChild(div).appendChild(img);
+  main.appendChild(fragmentLoadingImage);
+}
+
+const removeLoadingImage = () => {
+  const loadingImage = document.getElementById("js-loading");
+  main.removeChild(loadingImage);
+}
+
+const getSlideData = () => {
+  return new Promise((resolve) => {
+    try {
+      setTimeout(()=>{
+        resolve(fetch(url));
+      },3000
```

Comment:
IMO

`fetch` doesn't reject the response if an error occurs on the server side, so I think checking `response` here is would be more safe implementation :)
[MDN reference](https://developer.mozilla.org/ja/docs/Web/API/Fetch_API/Using_Fetch#%E3%83%95%E3%82%A7%E3%83%83%E3%83%81%E3%81%8C%E6%88%90%E5%8A%9F%E3%81%97%E3%81%9F%E3%81%8B%E3%81%AE%E7%A2%BA%E8%AA%8D)

Reply:
[ecf5043](https://github.com/hippopo-moo/js-handson-work/pull/19/commits/ecf5043655746dcd96de5d6519d1cf1f111bd857)

Thanks for the comment!

I didn't have that perspective at all, so I learned a lot.
I tried to revise it while researching, but I'm not sure if I'm writing it right. It would be great if you could take a look at it again.

Reply:
I think it's ok! 👏 
It's just IMO though, I also thought about the other way. How about this? :)
I unified this part with async/await since they are used in the init function,  and cut out the function to check the response.

```
const getSlideData = () => {
  return new Promise((resolve) => {
    try {
      setTimeout(() => {
        resolve(fetchJson(url));
      }, 3000);
    } catch (error) {
      console.error(error);
    } finally {
      console.log("finally");
    }
  });
};

const fetchJson = async (jsonUrl) => {
  const response = await fetch(jsonUrl);
  if (!response.ok) {
    throw new Error(`Network response was not OK: ${response.status}`);
  }
  return response;
};
```

Reply:
[40c76ed](https://github.com/hippopo-moo/js-handson-work/pull/19/commits/40c76ed0472c632a109aad763d06a29ff8ca5b0d)

```
if (!response.ok) {
    throw new Error(`Network response was not OK: ${response.status}`);
  }
```
↑Sure, I figured it would be easier to see if I cut it out, so I fixed it!
It would be better and easier to see if we detect the case of error here first. Thanks for the new insight!